gym/lib/gym/runner.rb

```diff
@@ -72,6 +73,11 @@ def clear_old_files
       end
     end
 
+    def fix_archive
+      return unless Gym.config[:use_legacy_build_api]
```

Comment:
This is confusing, with a lot of `!`, `unless` and `nil?`. Could we split it up to multiple lines?